### ISSUE TITLE
EADDRINUSE not handled

### ISSUE THREAD

**Issue by bendavis78** [modelcontextprotocol/inspector#214]
If port 5173 is already in use, inspector will fail to run, but the CLI incorrectly reports the inspector is up and running.

**Steps to reproduce:**

1. Have something listen on port 5173, eg: `nc -lp 5173`
2. Run `mcp dev my-server.py`:

```sh
Need to install the following packages:
@modelcontextprotocol/inspector@0.6.0
Ok to proceed? (y)

Starting MCP inspector...
node:events:497
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use 0.0.0.0:5173
    at Server.setupListenHandle [as _listen2] (node:net:1904:16)
    at listenInCluster (node:net:1961:12)
    at Server.listen (node:net:2063:7)
    at file:///home/ben/.npm/_npx/5a9d879542beca3a/node_modules/@modelcontextprotocol/inspector/client/bin/cli.js:19:8
    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:123:5)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1940:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EADDRINUSE',
  errno: -98,
  syscall: 'listen',
  address: '0.0.0.0',
  port: 5173
}

Node.js v20.16.0
Proxy server listening on port 3000

🔍 MCP Inspector is up and running at http://localhost:5173 🚀
```

**Expected behavior**

The command should exit immediately on any raised error and output the error. It should not erroneously report that the inspector is "up and running".

**Comment by cliffhall** [comment]
Working on this...
